tools/kagami/build.rs

```diff
@@ -1,8 +1,12 @@
 //! Provides macros used to get the version
 
-use std::error::Error;
+use std::{env::set_var, error::Error};
 
 fn main() -> Result<(), Box<dyn Error>> {
+    // We default to iroha2-dev's latest hash to enable shallow cloning in swarm
+    let repo = gix::open("../..")?;
+    let sha = repo.rev_parse_single("iroha2-dev")?;
+    set_var("VERGEN_GIT_SHA", sha.to_string());
```

Comment:
Not sure if Swarm should bind to a particular branch (e.g. `iroha2-dev`, `iroha2-stable` etc). IMO, it makes more sense to bind to the same revision Kagami is built itself, whatever branch it is. Also, I think I wrote this in Kagami CLI help.

Reply:
I _think_ using `HEAD` as revision specification here would do the necessary thing

Reply:
The shallow git clone for the validator fails in case of using just `HEAD` as it tries to grab the wrong SHA - presumably on only this PR, but still. This whole contraption is there to circumvent it. I guess I'll have to go for `HEAD~1`. This is likely to break CI for PRs with many commits though..

Reply:
Yeah, that makes sense.. There's also `git merge-base` (don't know gix-speak for it). Though, I am not sure if using older version than the PR contains even makes sense for a CI check..